packages/mongodb-mql-model/src/main/kotlin/com/mongodb/jbplugin/mql/BsonType.kt

```diff
@@ -15,7 +15,14 @@ import java.util.*
 /**
  * Represents any of the valid BSON types.
  */
-interface BsonType
+interface BsonType {
+    fun isMatching(otherType: BsonType): Boolean = when (otherType) {
+        this -> true
+        is BsonAny -> true
+        is BsonAnyOf -> otherType.types.any { this.isMatching(it) }
+        else -> false
```

Comment:
I think the logic that is missing here is when `this is BsonAnyOf` something. I don't see the implementation in this fil.

Reply:
Seen it below, the diff was a bit weird :D.